pages/api/conversations/export.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable quotes */
+import { AgentVisibility, Conversation, Message } from "@prisma/client";
+import Cors from "cors";
+import JSZip from "jszip";
+import { NextApiResponse } from "next";
+import { z } from "zod";
+
+import { ConversationWithMessages } from "@app/types/dtos";
+import { AppNextApiRequest } from "@app/types/index";
+import { ApiError, ApiErrorType } from "@app/utils/api-error";
+import { createAuthApiHandler, respond } from "@app/utils/createa-api-handler";
+import { generateExcelBuffer } from "@app/utils/export/excel-export";
+import prisma from "@app/utils/prisma-client";
+import runMiddleware from "@app/utils/run-middleware";
+import validate from "@app/utils/validate";
+
+const handler = createAuthApiHandler();
+
+const cors = Cors({
+  methods: ["POST", "HEAD"],
+});
+
+async function exportConversation(
+  req: AppNextApiRequest,
+  res: NextApiResponse
+) {
+  const conversationIds = req.body.ids as string[];
+  const conversations = await prisma.conversation.findMany({
+    where: {
+      id: {
+        in: conversationIds,
+      },
+    },
+    include: {
+      messages: true,
+    },
+  });
+
+  const zipBuffer = await exportConversations(conversations);
+
+  res.setHeader("Content-Type", "application/zip");
+  res.setHeader("Content-Disposition", "attachment; filename=data.zip");
+  res.send(zipBuffer);
+}
+
+handler.post(
+  validate({
+    body: z.object({ ids: z.array(z.string().cuid()) }),
+    handler: respond(exportConversation),
+  })
+);
+
+async function exportConversations(conversations: ConversationWithMessages[]) {
+  const zip = new JSZip();
+
+  const header = [
+    "conversation__id",
+    "message_id",
+    "text",
+    "from",
+    "sources",
+    "evaluation",
+    "read",
+    "createdAt",
+  ];
+
+  const zipConversation = async (conversation: ConversationWithMessages) => {
+    let rows = [];
+    for (const message of conversation.messages) {
+      const messageDetails = [
+        message.conversationId,
+        message.id,
+        message.text,
+        message.from,
+        JSON.stringify(message.sources),
+        message.eval,
+        message.read,
+        message.createdAt,
+      ];
+      rows.push(messageDetails);
+    }
+    const excelBuffer = await generateExcelBuffer<Message>({ header, rows });
+    zip.file(`${conversation.id}.csv`, excelBuffer);
+  };
+
+  const zipPromises = conversations.map((conversation) =>
+    zipConversation(conversation)
+  );
+  await Promise.all(zipPromises);
+
+  return zip.generateAsync({ type: "nodebuffer" });
+}
+
+export default async function wrapper(
```

Comment:
Can you replace this code with "export default handler;" 

This wrapper is only necessary for endpoints available through our official Chaindesk API (to authorize external domain to call the endpoint).

This endpoint is not meant to be available through the Chaindesk API